Cython/Compiler/Parsing.py

```diff
@@ -2967,7 +2967,17 @@ def p_c_func_declarator(s, pos, ctx, base, cmethod_flag):
     ellipsis = p_optional_ellipsis(s)
     s.expect(')')
     nogil = p_nogil(s)
-    exc_val, exc_check = p_exception_value_clause(s)
+    if nogil:
+        warning(
+            s.position(),
+            "The keyword 'nogil' must appear at the end of the "
+            "function signature line. Placing it before 'except' "
+            "or 'noexcept' will be disallowed in a future version "
+            "of Cython.",
+            level=1
```

Comment:
`level=1` doesn't get shown by default. If we're serious about disallowing it in a future version of Cython we should probably make it `level=2`. (If we're not serious then `level=1` is OK)

I also wonder if "must" -> "should". It's a little gentler working for something we don't yet enforce.